R/bm-tpc-h.R

```diff
@@ -639,6 +639,83 @@ tpc_h_queries[[10]] <- function(input_func) {
     collect()
 }
 
+tpc_h_queries[[11]] <- function(input_func) {
+  # SELECT
+  #     ps_partkey,
+  #     sum(ps_supplycost * ps_availqty) AS value
+  # FROM
+  #     partsupp,
+  #     supplier,
+  #     nation
+  # WHERE
+  #     ps_suppkey = s_suppkey
+  #     AND s_nationkey = n_nationkey
+  #     AND n_name = 'GERMANY'
+
+  partsupp <- input_func("partsupp")
+  supplier <- input_func("supplier")
+  nation <- input_func("nation")
+
+  joined_filtered <- partsupp %>%
+    inner_join(supplier, by = c("ps_suppkey" = "s_suppkey")) %>%
+    inner_join(nation, by = c("s_nationkey" = "n_nationkey")) %>%
+    filter(n_name == "GERMANY")
+
+  # GROUP BY
+  #     ps_partkey
+  # HAVING
+  #     sum(ps_supplycost * ps_availqty) > (
+  #         SELECT
+  #             sum(ps_supplycost * ps_availqty) * 0.0001000000
+  #         FROM
+  #             partsupp,
+  #             supplier,
+  #             nation
+  #         WHERE
+  #             ps_suppkey = s_suppkey
+  #             AND s_nationkey = n_nationkey
+  #             AND n_name = 'GERMANY')
+  # ORDER BY
+  #     value DESC;
+
+  # grouped mutate/filter works in dplyr but errors in Arrow because
+  # window functions aren't currently supported
+  # result1 <- joined_filtered %>%
+  #   mutate(global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000) %>%
+  #   group_by(ps_partkey) %>%
+  #   mutate(value = sum(ps_supplycost * ps_availqty)) %>%
+  #   filter(value > global_value) %>%
+  #   ungroup() %>%
+  #   arrange(desc(value)) %>%
+  #   select(-global_value) %>%
+  #   collect()
+
+  # ...but not in Arrow yet, so we do a summarize() + left_join()
+  global_agr <- joined_filtered %>%
+    mutate(global_agr_key = 1L) %>%
+    group_by(global_agr_key) %>%
+    summarise(
+      global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000
+    )
+
+  partkey_agr <- joined_filtered %>%
+    group_by(ps_partkey) %>%
+    summarise(value = sum(ps_supplycost * ps_availqty))
+
+  result2 <- joined_filtered %>%
+    mutate(global_agr_key = 1L) %>%
+    left_join(global_agr, by = "global_agr_key") %>%
```

Comment:
This _should_ work. I took [a slightly different approach](https://github.com/ursacomputing/arrowbench/pull/51/files#diff-b47df652daa060d3d75dd06b6b88ea3772a7e83a16a15db1af70ce36017db401R678-R690) where I `collect()`ed the global aggregate and then pulled out the only element from it (I could and probably should do that in a slightly less fragile way). 

Something we might want to make a Jira for (though there might be one already or this might be implemented and I just didn't have the right call at my fingertips!) is if we can `pull()` a single value out of a query like that which would let us not `collect()` the global aggregate at first.